cmd2/command_definition.py

```diff
@@ -0,0 +1,131 @@
+# coding=utf-8
+"""
+Supports the definition of commands in separate classes to be composed into cmd2.Cmd
+"""
+import functools
+from typing import (
+    Callable,
+    Dict,
+    Iterable,
+    Optional,
+    Type,
+    Union,
+)
+from .constants import COMMAND_FUNC_PREFIX
+
+# Allows IDEs to resolve types without impacting imports at runtime, breaking circular dependency issues
+try:  # pragma: no cover
+    from typing import TYPE_CHECKING
+    if TYPE_CHECKING:
+        from .cmd2 import Cmd, Statement
+        import argparse
+except ImportError:
```

Comment:
When would this `ImportError` ever occur?  `TYPE_CHECKING` was added in 3.5.2.  Theoretically we say we support Python `3.5+`.  But I'd be OK bumping that to `3.5.2+` if need be.

Reply:
That's the only reason for this check, for python earlier than 3.5.2.